### PR TITLE
feat: configure-bridge .sh

### PR DESCRIPTION
**Pull request by 0xmovses**
# Summary
Creates a script for configuring the bridge relayer once it's running. Boolean arg sets whether move scripts should be run, as these are only required to be run once. 

Note that if they are run a second time on the same network an error will be thrown:
```
"Error": "Simulation failed with status: Move abort in 0x1::transaction_fee: ECOPY_CAPS_SHOT(0x7)
```
This is expected and it indicates the the user already has the mint and burn capabilities. 

# Testing
1. Local suzuka-node should be running
2. At workspace root run 
`./scripts/services/bridge/configure-bridge.sh true`

### CODE REVIEW

File: scripts/services/bridge/configure-bridge.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    # Linux (GNU sed)
+    sed -i "s/^    account: .*/    account: ${NEW_ACCOUNT}/" "$CONFIG_FILE"
+fi
+
+echo "Account field updated with value: ${NEW_ACCOUNT}"
+
+# Execute the Move scripts if execute_move is true
+if [ "$EXECUTE_MOVE" == "true" ]; then
+  echo "Executing Move scripts..."
+
```
Comment (andygolay) [inline scripts/services/bridge/configure-bridge.sh:43]:
```suggestion
  movement move compile \ 
    --package-dir protocol-units/bridge/move-modules/
    --profile default \
    --assume-yes 
```

**Comment by andygolay** [comment]
When I ran 

```
./scripts/services/bridge/configure-bridge.sh true
```
against the node, I got this:

```
 ./scripts/services/bridge/configure-bridge.sh true   
Account field updated with value: 0xA550C18
Executing Move scripts...
{
  "Error": "Error accessing 'Unable to read Some(\"protocol-units/bridge/move-modules/build/bridge-modules/bytecode_scripts/enable_bridge_feature.mv\")': No such file or directory (os error 2)"
}
{
  "Error": "Error accessing 'Unable to read Some(\"protocol-units/bridge/move-modules/build/bridge-modules/bytecode_scripts/store_mint_burn_caps.mv\")': No such file or directory (os error 2)"
}
Move scripts executed.
```

Added the compile step here as a suggestion to fix it: https://github.com/movementlabsxyz/movement/pull/737/files#r1813018755

File: scripts/services/bridge/configure-bridge.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+echo "Account field updated with value: ${NEW_ACCOUNT}"
+
+# Execute the Move scripts if execute_move is true
+if [ "$EXECUTE_MOVE" == "true" ]; then
+  echo "Executing Move scripts..."
+  movement move compile \ 
+    --package-dir protocol-units/bridge/move-modules/
+    --profile default \
+    --assume-yes 
+  # First script: enable_bridge_feature
```
Comment (andygolay) [inline scripts/services/bridge/configure-bridge.sh:47]:
```suggestion
    --assume-yes 
    
  # First script: enable_bridge_feature
```

**Comment by 0xmovses** [comment]
Ah yes, probably they were already compiled in my build/ dir. I added the the compile step you suggested 👍

**Comment by andygolay** [comment]
> When I ran
> 
> ```
> ./scripts/services/bridge/configure-bridge.sh true
> ```
> 
> against the node, I got this:
> 
> ```
>  ./scripts/services/bridge/configure-bridge.sh true   
> Account field updated with value: 0xA550C18
> Executing Move scripts...
> {
>   "Error": "Error accessing 'Unable to read Some(\"protocol-units/bridge/move-modules/build/bridge-modules/bytecode_scripts/enable_bridge_feature.mv\")': No such file or directory (os error 2)"
> }
> {
>   "Error": "Error accessing 'Unable to read Some(\"protocol-units/bridge/move-modules/build/bridge-modules/bytecode_scripts/store_mint_burn_caps.mv\")': No such file or directory (os error 2)"
> }
> Move scripts executed.
> ```
> 
> Added the compile step here as a suggestion to fix it: https://github.com/movementlabsxyz/movement/pull/737/files#r1813018755

Looks like my change suggestion may have gotten put in as a commit, or you're really fast @0xmovses . Added one suggested nit for spacing: https://github.com/movementlabsxyz/movement/pull/737/files#r1813025522

**Comment by 0xmovses** [comment]
@andygolay when you accept the suggested change from within GH UI, it adds it as a commit from you, so the PR becomes co-authored.

**Comment by andygolay** [comment]
> @andygolay when you accept the suggested change from within GH UI, it adds it as a commit from you, so the PR becomes co-authored.

I didn't accept it though, only made it as a suggestion so it seems strange for it to have been added as a commit unless you accepted it immediately after I made it. 

With that update I'm getting `./scripts/services/bridge/configure-bridge.sh: line 44: --package-dir: command not found`, must be some typo; that is the name of the option in `movement` CLI... will test locally

**Comment by 0xmovses** [comment]
I accepted after they came in. No they shouldn't be compiled anywhere. I can check the error, jumping between PRs a bit atm.

File: scripts/services/bridge/configure-bridge.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+echo "Account field updated with value: ${NEW_ACCOUNT}"
+
+# Execute the Move scripts if execute_move is true
+if [ "$EXECUTE_MOVE" == "true" ]; then
+  echo "Executing Move scripts..."
+  movement move compile \ 
+    --package-dir protocol-units/bridge/move-modules/
+    --profile default \
+    --assume-yes
```
Comment (andygolay) [inline scripts/services/bridge/configure-bridge.sh]:
```suggestion
  movement move compile \
    --package-dir protocol-units/bridge/move-modules/
```

**Comment by andygolay** [comment]
> I accepted after they came in. No they shouldn't be compiled anywhere. I can check the error, jumping between PRs a bit atm.

Fixed here: https://github.com/movementlabsxyz/movement/pull/737/files#r1813082455

Tested locally, got to Move scripts executed.

It needs update bridge operator as well, right? If the operator is going to be the `f9..` address? I'm not sure what the plan is for what address the operator is intended to be or how that will be updated.

**Comment by 0xmovses** [comment]
Updated I now get this 
```
{
  "Result": [
    "b33977f2c181730ab53211a00f0db4fda260797954ff8a73edca780e56fc8f44::moveth_tests",
    "b33977f2c181730ab53211a00f0db4fda260797954ff8a73edca780e56fc8f44::moveth",
    "b33977f2c181730ab53211a00f0db4fda260797954ff8a73edca780e56fc8f44::atomic_bridge_counterparty",
    "b33977f2c181730ab53211a00f0db4fda260797954ff8a73edca780e56fc8f44::atomic_bridge_initiator",
    "b33977f2c181730ab53211a00f0db4fda260797954ff8a73edca780e56fc8f44::move_fa"
  ]
}
Transaction submitted: https://explorer.movementlabs.xyz/txn/0xf8227034148375af5660a96e04bccadfabce02a4634b84e041853057ca52d16e?network=custom
{
  "Result": {
    "transaction_hash": "0xf8227034148375af5660a96e04bccadfabce02a4634b84e041853057ca52d16e",
    "gas_used": 7,
    "gas_unit_price": 100,
    "sender": "000000000000000000000000000000000000000000000000000000000a550c18",
    "sequence_number": 19,
    "success": true,
    "timestamp_us": 1729698776483003,
    "version": 82,
    "vm_status": "Executed successfully"
  }
}
{
  "Error": "Simulation failed with status: Move abort in 0x1::transaction_fee: ECOPY_CAPS_SHOT(0x7): "
}
Move scripts executed.
```

**Comment by 0xmovses** [comment]
Ah yes I will add the update bridge operator. The bridge operator address will be updated by running this script, the value of that should be the equivalent of what we use locally. I'll DM you.

File: scripts/services/bridge/configure-bridge.sh [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  # Second script: store_mint_burn_caps
+  movement move run-script \
+    --compiled-script-path protocol-units/bridge/move-modules/build/bridge-modules/bytecode_scripts/store_mint_burn_caps.mv \
+    --profile default \
+    --assume-yes 2>&1 | tee store_mint_burn_caps_output.log
+
+  # Third script: update_bridge_operator 
+  movement move run-script \
+    --compiled-script-path protocol-units/bridge/move-modules/build/bridge-modules/bytecode_scripts/update_bridge_operator.mv \
+    --args address:${BRIDGE_OPERATOR_ADDRESS} \
```
Comment (andygolay) [inline scripts/services/bridge/configure-bridge.sh:63]: So `BRIDGE_OPERATOR_ADDRESS` is passed in as an env var here?

Or would it make sense to hard-code in the `f9...` address?

**Comment by l-monninger** [comment]
Is this script only supposed to be run once? Does it require that the older version of the contracts were deployed before?

**Comment by 0xmovses** [comment]
@l-monninger The move scripts that this calls `enable_bridge_feature`, `store_mint_burn_caps`, `update_bridge_operator`, don't require any deployment of move modules as they interact with the framework itself. `aptos-framework`, rev value has been bumped into main to include the related code. 

Yes it should only be called once, however if it is called a second time no write actions occur, nothing terrible happens.

**Review by l-monninger — APPROVED** [review]

**Pull request merged**